src/ILLink.Shared/TrimAnalysis/HandleCallAction.cs

```diff
@@ -38,6 +45,89 @@ public bool Invoke (MethodProxy calledMethod, MultiValue instanceValue, IReadOnl
+				foreach (var value in instanceValue) {
+					if (value is SystemTypeValue typeValue)
+						methodReturnValue = MultiValueLattice.Meet (methodReturnValue, new RuntimeTypeHandleValue (typeValue.RepresentedType));
+					else if (value is GenericParameterValue genericParameterValue)
+						methodReturnValue = MultiValueLattice.Meet (methodReturnValue, new RuntimeTypeHandleForGenericParameterValue (genericParameterValue.GenericParameter));
+					else if (value == NullValue.Instance)
+						methodReturnValue = MultiValueLattice.Meet (methodReturnValue, value);
+					else
+						methodReturnValue = MultiValueLattice.Meet (methodReturnValue, GetMethodReturnValue (calledMethod, returnValueDynamicallyAccessedMemberTypes));
+				}
+				break;
+
+			//
+			// GetInterface (String)
+			// GetInterface (String, bool)
+			//
+			case IntrinsicId.Type_GetInterface: {
+					var targetValue = GetMethodThisParameterValue (calledMethod, DynamicallyAccessedMemberTypesOverlay.Interfaces);
+					foreach (var value in instanceValue) {
+						// For now no support for marking a single interface by name. We would have to correctly support
+						// mangled names for generics to do that correctly. Simply mark all interfaces on the type for now.
+
+						// Require Interfaces annotation
+						_requireDynamicallyAccessedMembersAction.Invoke (diagnosticContext, value, targetValue);
+
+						// Interfaces is transitive, so the return values will always have at least Interfaces annotation
+						DynamicallyAccessedMemberTypes returnMemberTypes = DynamicallyAccessedMemberTypesOverlay.Interfaces;
+
+						// Propagate All annotation across the call - All is a superset of Interfaces
+						if (value is ValueWithDynamicallyAccessedMembers valueWithDynamicallyAccessedMembers
+							&& valueWithDynamicallyAccessedMembers.DynamicallyAccessedMemberTypes == DynamicallyAccessedMemberTypes.All)
+							returnMemberTypes = DynamicallyAccessedMemberTypes.All;
+
+						methodReturnValue = MultiValueLattice.Meet (methodReturnValue, GetMethodReturnValue (calledMethod, returnMemberTypes));
+					}
+				}
+				break;
+
+			//
+			// AssemblyQualifiedName
+			//
+			case IntrinsicId.Type_get_AssemblyQualifiedName: {
+					foreach (var value in instanceValue) {
+						if (value is ValueWithDynamicallyAccessedMembers valueWithDynamicallyAccessedMembers) {
+							// Currently we don't need to track the difference between Type and String annotated values
+							// that only matters when we use them, so Type.GetType is the difference really.
+							// For diagnostics we actually don't want to track the Type.AssemblyQualifiedName
+							// as the annotation does not come from that call, but from its input.
+							methodReturnValue = MultiValueLattice.Meet (methodReturnValue, valueWithDynamicallyAccessedMembers);
+						} else if (value == NullValue.Instance) {
+							// Just propagate null - it's not perfect, but we can't just skip it, we have to produce some value to the output
```

Comment:
What's wrong with just skipping it?

Reply:
Currently there's a technical problem - if the only input is `null`, and we skip it, then we produce empty methodReturnValue. The code below tries to handle cases where the intrinsics doesn't do anything, and thus if the return value is empty, it will replace it with the "return value of the method". Which is effectively unknown value which could lead to warnings.

We would have to either somehow remember that we did "set" the return value, make it nullable for example, or modify the code so that intrinsics never "not return value" in which case we could remove the code below.

Reply:
I see, thanks. I would personally prefer to remember whether the return value was set (I like the nullable suggestion) but am also ok with the workaround.

Reply:
I tried, but it causes problems in the linker. Currently it's pretty tricky - linker is not setup to return "empty value". If that happens the `MethodBodyScanner` basically fails thinking it was an invalid IL (since it expects a real value on the stack).
So I had to keep the behavior - I did cleanup return value handling a little bit, but not functionality.